cmd/manager/main.go

```diff
@@ -225,3 +241,78 @@ func checkIfRequiredFilesExist(requiredFiles []string) error {
 	}
 	return nil
 }
+
+// clusterConfigChecks checks for required configuration in a cluster before starting the controller.
+// If the cluster doesn`t meet the configuration, it returns an error.
+func clusterConfigChecks(cfg *rest.Config) error {
+	// Get openshift api config client.
+	clienset, err := configclient.NewForConfig(cfg)
+	if err != nil {
+		return fmt.Errorf("error creating clientset %v", err)
+	}
+
+	// Check for valid K8s version. Currently any version >= 1.18 is supported.
+	// Check for Openshift version is not required.
+	if err := isValidKubernetesVersion(clienset); err != nil {
+		return fmt.Errorf("error in k8s version of the cluster %v", err)
+	}
+	// Check for determining if the cluster uses ovn-kubernetes
+	if err = isValidNetworkType(clienset); err != nil {
+		return fmt.Errorf("error in network type of the cluster %v", err)
+	}
+	//Check for determining if hybrid overlay is enabled
+	if err = isHybridOverlayEnabled(cfg); err != nil {
+		return fmt.Errorf("error in hybrid overlay %v", err)
+	}
+
+	return nil
+}
+
+// isValidKubernetesVersion checks for valid k8s version in the cluster. It returns an err for all versions below
+// v1.18.0
+func isValidKubernetesVersion(clientset *configclient.Clientset) error {
+	versionInfo, err := clientset.Discovery().ServerVersion()
+	if err != nil {
+		return err
+	}
+
+	k8sVersion := versionInfo.GitVersion
+	if strings.Compare(k8sVersion, baseK8sVersion) < 0 {
```

Comment:
Thinking more about this... Do we want to be more specific given that we will be cutting different operators for different versions of OpenShift? This would allow an user to run a 4.6 operator on a 4.5 cluster and vice versa. Please bring this up at the next stand up.